### DISCUSSION TITLE
Github Page Deploy Keeps Failing

### DISCUSSION THREAD

**Discussion by JerryCG** [community/community#30907] [Other Feature Feedback, Questions, & Ideas]
After I pushed the commit to Github repository, the Github Pages began to build. The webpage can be successfully built but cannot be deployed. I have re-run the failed deploy task several times but cannot make it through. I see the log files, it seems to be stuck in the "current status". 
<img width="859" alt="fail" src="https://user-images.githubusercontent.com/66578530/186883406-ce039c4f-c6b8-4c49-ac15-1cb1e30b31a6.png">

**Comment by yoannchaudet** [comment]
You are trying to deploy a very large site:

<img width="525" alt="image" src="https://user-images.githubusercontent.com/14911070/186969278-b83921d3-8906-4605-9ee4-ec159b208320.png">

It's 5GB compressed and way over our [1GB limit](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#usage-limits). There are better services out there for uploading music collections.

**Comment by JerryCG** [comment]
Thanks. I didn't realize there is a limit to repository size. But previously I could successfully deploy the pages XD. Btw, what are your suggestion for hosting a music collection webpage like this?

**Comment by ichenglin** [comment]
> **Message Quoted from [JerryCG](https://github.com/JerryCG)**. **[(view original)](https://github.com/orgs/community/discussions/30907#discussioncomment-3487411)**
> Thanks. I didn't realize there is a limit to repository size. But previously I could successfully deploy the pages XD. Btw, what are your suggestion for hosting a music collection webpage like this?

I think there's some confusion here,
in which **[JerryCG](https://github.com/JerryCG)** believes that the **web page has to be hosted together with the music files** (which took up most of the space).
But in reality, you could **separate the two** (web page & music files) and host them separately on different services.

For example (and my recommendation),
you could still **deploy your web page** source code on **Github** (for its Github Pages features),
while having the **web page referring** (importing from) to **third-party services** for music (such as **Youtube**, free storage why not).

**Reply by JerryCG** [reply]
Oh, your suggestion is very constructive. Thanks so much!
